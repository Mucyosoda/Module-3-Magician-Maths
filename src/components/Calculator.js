import React, { Component } from 'react';

// eslint-disable-next-line react/prefer-stateless-function
class Calc extends Component {
  render() {
    return (
      <div className="App">
        <div className="calculator">
          <div className="display">0</div>
          <div className="display-buttons">
            <div className="operators">
              <button type="button">รท</button>
              <button type="button">x</button>
              <button type="button">-</button>
              <button type="button">+</button>
              <button type="button" id="equal">=</button>
            </div>

            <div className="digits">
              <button type="button" className="btn">
                AC
              </button>
              <button type="button" className="btn">
                +/-
              </button>
              <button type="button" className="btn">
                %
              </button>

              <button type="button" className="btn">
                7
              </button>
              <button type="button" className="btn">
                8
              </button>
              <button type="button" className="btn">
                9
              </button>
              <button type="button" className="btn">
                4
              </button>
              <button type="button" className="btn">
                5
              </button>

              <button type="button" className="btn">
                6
              </button>
              <button type="button" className="btn">
                1
              </button>
              <button type="button" className="btn">
                2
              </button>
              <button type="button" className="btn">
                3
              </button>
              <button type="button" className="zero">
                0
              </button>

              <button type="button" className="point">
                .
              </button>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Calc;
