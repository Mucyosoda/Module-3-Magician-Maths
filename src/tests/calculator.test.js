import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../Components/Calculator';

it('calculator snapshot test', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
