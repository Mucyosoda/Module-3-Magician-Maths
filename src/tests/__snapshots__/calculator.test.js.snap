// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculator snapshot test 1`] = `
<div>
  <h3
    className="mathshead"
  >
    Calculator
  </h3>
  <div
    className="calc-container"
  >
    <div>
      <div
        className="calc-screen"
      >
        
      </div>
      <div
        className="buttons"
      >
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="orange-button"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="orange-button"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="orange-button"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="orange-button"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="big-space"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="orange-button"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
